gprofiler/merge.py

```diff
@@ -35,7 +35,7 @@
 # ffffffff81082227 mmput+0x57 ([kernel.kallsyms])
 # 0 [unknown] ([unknown])
 # 7fe48f00faff __poll+0x4f (/lib/x86_64-linux-gnu/libc-2.31.so)
-FRAME_REGEX = re.compile(r"^\s*[0-9a-f]+ (.*?) \((.*)\)$")
+FRAME_REGEX = re.compile(r"^\s*[0-9a-f]+ (.*?) \(\[?(.*?)\]?\)$")
```

Comment:
You changed the 2nd capture group here but didn't update it accordingly in `_collapse_stack` (i.e the removal of brackets).

Personally, I like writing unit tests for such functions, so that we can test the entire set of finite cases of behavior for it. I recently did the same for `test_get_average_frame_count`. Let's add the same for `_collapse_stack` and add tests that follow the conformance I suggested [here](https://github.com/Granulate/gprofiler/pull/491#issuecomment-1258187962)?

Reply:
Thanks for suggestion.